src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -22,6 +44,31 @@ const freeformFilters: FilterDefinition<
       return (loadout) => test(loadout.name);
     },
   },
+  {
+    keywords: ['subclass'],
+    description: tl('LoadoutFilter.Name'),
+    format: ['simple', 'freeform'],
+    suggestionsGenerator: ({ loadouts, d2Manifest }) =>
+      loadouts && d2Manifest
+        ? Array.from(
+            new Set(
+              subclassesFromLoadouts(loadouts, d2Manifest).map(
+                (subclass) =>
+                  // TODO (ryan) subclasses have a none damage type, so to do subclass match
+                  // based on element name (solar/stasis/etc) we need to set up some known data
+                  `subclass:${quoteFilterString(subclass.displayProperties.name.toLowerCase())}`,
+              ),
+            ),
+          )
```

Comment:
```suggestion
        ? Array.from(
            new Set(loadouts.map((l) => subclassFromLoadout(l, d2Manifest).filter((s) => s !== undefined))
            ),
          ).map(
            (subclass) =>
              // TODO (ryan) subclasses have a none damage type, so to do subclass match
              // based on element name (solar/stasis/etc) we need to set up some known data
                         `subclass:${quoteFilterString(subclass.displayProperties.name.toLowerCase())}`,
          ),
```

Since the definitions are reference-identical, you can crunch the set first, then map to strings. You can also avoid needing to define a second `subclassesFromLoadouts` function.

Reply:
I was getting type errors of undefined even with a filter function. I was using `Boolean` so maybe it was just that.

Reply:
Hm I guess automatic type predicates are in TypeScript 5.5. They explain why even then, you can't use `Boolean`: https://devblogs.microsoft.com/typescript/announcing-typescript-5-5-beta/#inferred-type-predicates

You can use `_.compact` in the meantime.

Reply:
I added `(s): s is Type` but can move to compact. It will probably be nicer to go through and cleanup the use of compact in preference to filters when we are on 5.5

Reply:
Move to compact done